### PR TITLE
[FIX] web: adapted Dropdown component documentation to new api

### PR DESCRIPTION
**Pull request by BastienFafchamps**
This commit adapts the documentation to reflect changes made to the Dropdown component after its refactor, namely:
- Adapted props list
- Adapted code snippets
- Added DropdownGroup component documentation
- More consise examples

task-id: 3266145

### CODE REVIEW

**Review by fdardenne — COMMENTED** [review]
Well written, good jobs 👍 
Few minor points just for the sake of consistency

File: content/developer/reference/frontend/owl_components.rst [hunk 1]
```diff
@@ -420,271 +420,277 @@ Location
...
+      - `"none"` | `"closest"` | `"all"`
+      - Optional, controls which parent dropdown should close when the item is selected:
+        `none`: the dropdown will not close, `closest`: the direct parent will close, `all`: every nested parent dropdown will close (default: `all`)
+    * - `attrs`
+      - `Object`
+      - Optional object representing attributes that are added to the root element. `<DropdownItem attrs="{ title: 'A tooltip', 'data-hotkey': 'shift+a' }">`. (If `href` is set, the element will automatically become an `a` element).
+
+.. important::
+   When writing custom css for you components, do not forget that the menu elements are not next to the toggle
+   but inside the overlay container, at the bottom of the document. Thus, **use the `menuClass` and `class` props to more
```
Comment (AntoineVDV) [inline content/developer/reference/frontend/owl_components.rst:539]: Markup within markup doesn't work in RST.

**Review by AntoineVDV — APPROVED** [review]
@robodoo delegate+

**Comment by BastienFafchamps** [comment]
@robodoo r+

**Pull request closed without merging**